src/main/java/loanbook/logic/commands/SummaryCommand.java

```diff
@@ -0,0 +1,100 @@
+package loanbook.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static loanbook.model.Model.PREDICATE_SHOW_ALL_LOANS;
+
+import java.util.List;
+
+import loanbook.logic.CommandHistory;
+import loanbook.model.Model;
+import loanbook.model.loan.Loan;
+import loanbook.model.loan.LoanStatus;
+
+/**
+ * Terminates the program.
+ */
+public class SummaryCommand extends Command {
+
+    public static final String COMMAND_WORD = "summary";
+
+    public static final String MESSAGE_SUMMARY_ACKNOWLEDGEMENT = "Summarized LoanBook!\n%1$s";
+
+    @Override
+    public CommandResult execute(Model model, CommandHistory history) {
+        requireNonNull(model);
+        model.updateFilteredLoanList(PREDICATE_SHOW_ALL_LOANS);
+
+        Summary summary = new Summary();
+
+        List<Loan> lastShownList = model.getFilteredLoanList();
+
+        for (Loan loan : lastShownList) {
+            summary.addLoan(loan);
+        }
+
+        return new CommandResult(String.format(MESSAGE_SUMMARY_ACKNOWLEDGEMENT, summary.getSummary()));
+    }
+}
+
+/**
+ * Class to encapsulate all the statistics to be kept track of.
+ */
+class Summary {
```

Comment:
I think it might be a good idea to have the `Summary` class in its own file.

Otherwise, you can possibly have it as a static nested class within the `SummaryCommand` class, similar to `EditLoanDescriptor` in the `EditLoan` class.